client/verta/verta/registry/_validate_input.py

```diff
@@ -0,0 +1,35 @@
+import functools
+import json
+import os
+
+import jsonschema
+
+# for use like: `if getattr(model.predict, _DECORATED_FLAG, False)`
+_VALIDATE_DECORATED_FLAG = "_verta_validate_input"
+_MODEL_SCHEMA_PATH = os.environ.get("VERTA_MODEL_SCHEMA_PATH", "/app/model_schema.json")
+
+
+def validate_input(f):
+    """Decorator to validate prediction input against previously provided schema."""
+
+    @functools.wraps(f)
+    def wrapper(self, *args, **kwargs):
+        # accepts dict input
+        # TODO: validate length of args and type
+        prediction_input = args[0]
+        try:
+            with open(_MODEL_SCHEMA_PATH, "r") as file:
+                # Load the JSON data into a variable
+                schema = json.load(file)
+        except FileNotFoundError:
+            raise Exception(
+                "No schema found for model. Did you remember to `set_schema`?"
+            )
```

Comment:
- we should raise `FileNotFoundError` instead of `Exception` to be as specific as possible
- Python's standard library has this weird practice of exception messages _not_ being capitalized, so it's dumb but I'm nudging the client to use lowercase and semicolons 🤷 
- we should raise `from e`; this tells Python "I'm deliberately handling the underlying error as a new error" instead of "oh no an unexpected new error ? ? ?"

```suggestion
        except FileNotFoundError as e:
            raise FileNotFoundError(
                "no schema found for model; did you remember to `set_schema`?"
            ) from e
```